e2e-cypress/files/pdf-generator.ts

```diff
@@ -0,0 +1,68 @@
+import * as puppeteer from 'puppeteer';
+import * as path from 'path';
+import * as fs from 'fs';
+
+async function expandReportTestCases(htmlPage: puppeteer.Page) {
+  await htmlPage.evaluate(() => {
+    const expandTestHeaders = document.querySelectorAll('[class^="test--header"], [class*="test--header"]');
+    for (const testHeader of Array.from(expandTestHeaders)) {
+      (testHeader as HTMLElement).click();
+    }
+  });
+}
+
+const isLocal = process.env.NODE_ENV === 'local';
+
+(async () => {
+  try {
+    const files = await fs.promises.readdir('build/test-results/mochawesome/');
+
+    for (const file of files) {
+
+      if (!fs.existsSync(path.resolve(__dirname, 'build/test-results/mochawesome/', 'pdf')))
+        fs.mkdirSync(path.resolve(__dirname, 'build/test-results/mochawesome/', 'pdf'));
```

Comment:
Creating a new folder should move to the spot where it is first needed. This would prevent situations where a new folder is created without further content, e.g. when the loop ends earlier due to the `continue` in the next lines or a runtime error.

Reply:
True, I moved it in my last push. I think I addressed all your comments with it. Thanks for all of them!